pkg/multimodelconfig/configmap.go

```diff
@@ -0,0 +1,112 @@
+package multimodelconfig
+
+import (
+	"encoding/json"
+	"k8s.io/api/core/v1"
+	"sigs.k8s.io/controller-runtime/pkg/runtime/log"
+)
+
+var logger = log.Log.WithName("MultiModelConfig")
+
+type ConfigsDelta struct {
+	updatedCfg map[string]string
+	deletedCfg map[string]string
+}
+
+type ModelConfig struct {
+	fileName    string
+	fileContent *ModelDefinition
+}
+
+//ModelDefinition will be replaced by ModelSpec
+//after we merge https://github.com/kubeflow/kfserving/pull/991 to master
+type ModelDefinition struct {
+	StorageUri string `json:"storageUri"`
+	Framework  string `json:"framework"`
+}
+
+func NewConfigsDelta(updatedCfg[]ModelConfig, deletedCfg []ModelConfig) (*ConfigsDelta, error) {
+	updatedcfgs, err := convert(updatedCfg...)
+	if err != nil {
+		return nil, err
```

Comment:
Consider annotating your error message e.g.
```return nil, fmt.Errorf("while parsing configuration to map, %v, err)```
This will give you nicely annotated error messages if you follow this pattern all the way up the stack.

"Failed to do something, while doing something else, while making an api call, while parsing the json"